isis/src/control/objs/BundleUtilities/BundleObservation.cpp

```diff
@@ -783,6 +784,366 @@ namespace Isis {
+    // instrumentPointing's center angles for RA, DEC, and TWI
+    if (nPointingCoefficients == 0) {
+      nPointingCoefficients = 1;
+      useDefaultPointing = true;
+    }
+
+    // Force number of position and pointing parameters to each be 3 (X,Y,Z; RA,DEC,TWI)
+    // so we can always output a value for them
+    int nPositionParameters = 3 * nPositionCoefficients;
+    int nPointingParameters = 3 * nPointingCoefficients;
+    int nParameters = nPositionParameters + nPointingParameters;
+
+    coefX.resize(nPositionCoefficients);
+    coefY.resize(nPositionCoefficients);
+    coefZ.resize(nPositionCoefficients);
+    coefRA.resize(nPointingCoefficients);
+    coefDEC.resize(nPointingCoefficients);
+    coefTWI.resize(nPointingCoefficients);
+
+    if (m_instrumentPosition) {
+      if (!useDefaultPosition) {
+        m_instrumentPosition->GetPolynomial(coefX, coefY, coefZ);
+      }
+      // Use the position's center coordinate if not solving for spacecraft position
+      else {
+        const std::vector<double> centerCoord = m_instrumentPosition->GetCenterCoordinate();
+        coefX[0] = centerCoord[0];
+        coefY[0] = centerCoord[1];
+        coefZ[0] = centerCoord[2];
+      }
+    }
+
+    if (m_instrumentRotation) {
+      if (!useDefaultPointing) {
+        m_instrumentRotation->GetPolynomial(coefRA, coefDEC, coefTWI);
+
+      }
+      // Use the pointing's center angles if not solving for pointing (rotation)
+      else {
+        const std::vector<double> centerAngles = m_instrumentRotation->GetCenterAngles();
+        coefRA[0] = centerAngles[0];
+        coefDEC[0] = centerAngles[1];
+        coefTWI[0] = centerAngles[2];
+
+      }
+    }
+
+    // for convenience, create vectors of parameters names and values in the correct sequence
+
+    QStringList parameterNamesListX,parameterNamesListY,parameterNamesListZ,
+        parameterNamesListRA,parameterNamesListDEC,parameterNamesListTWI,
+        parameterNamesList;
+    QStringList correctionUnitListX,correctionUnitListY,correctionUnitListZ,
+        correctionUnitListRA,correctionUnitListDEC,correctionUnitListTWI,
+        correctionUnitList;
+
+    QString str("%1(%2)  ");
+
+    finalParameterValues.append(QVector<double>::fromStdVector(coefX));
+    finalParameterValues.append(QVector<double>::fromStdVector(coefY));
+    finalParameterValues.append(QVector<double>::fromStdVector(coefZ));
+
+
+
+          if (nPositionCoefficients > 0) {
+          for (int j = 0; j < nPositionCoefficients;j++) {
+            if (j == 0) {
+              parameterNamesListX.append(str.arg("  X  ").arg("km"));
+              parameterNamesListY.append(str.arg("  Y  ").arg("km"));
+              parameterNamesListZ.append(str.arg("  Z  ").arg("km"));
+              correctionUnitListX.append("m");
+              correctionUnitListY.append("m");
+              correctionUnitListZ.append("m");
+            } //end inner-if
+
+            else {
+              switch(j) {
```

Comment:
Why a switch statement here at all? Seems to mixing if/else with switch statement. Should be one or the other.

Reply:
@Kelvinrr It started out life as a longer switch-case statement but was then truncated.  I've refactored it as an if-else if-else statement.